Ava.Application/Bookings/Commands/RejectBookingCommand.cs

```diff
@@ -0,0 +1,52 @@
+﻿using Ava.Application.Constants;
+using Ava.Application.Models;
+using Ava.Infrastructure.Db;
+
+using MediatR;
+
+using Microsoft.EntityFrameworkCore;
+
+namespace Ava.Application.Bookings.Commands;
+
+public record RejectBookingCommand(Guid bookingId) : IRequest<Result>;
+
+public class RejectBookingCommandHandler : IRequestHandler<RejectBookingCommand, Result>
+{
+    private readonly AvaDbContext _context;
+    public RejectBookingCommandHandler(AvaDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<Result> Handle(RejectBookingCommand request, CancellationToken cancellationToken)
+    {
+        var booking = await _context.Bookings.Where(b => b.Id == request.bookingId).SingleOrDefaultAsync(cancellationToken);
+
+        if (booking == null)
+        {
+            return Result.Failure(BookingErrors.NotFound);
+        }
+
+        var therapistEntity = await _context.Therapists.Where(t => t.Id == booking.TherapistId).SingleOrDefaultAsync(cancellationToken);
+
+        if (therapistEntity == null)
+        {
+            return Result.Failure(TherapistErrors.NotFound);
+        }
+
+        var hasTherapistRole = await _context.UserRoles
+            .Where(r => r.UserId == therapistEntity.Id)
```

Comment:
Check on role is used in multiple places. Try to create a class that will have a method that checks the role. Instead of db query I would use Role claim from JWT token